lib/jnpr/junos/device.py

```diff
@@ -550,7 +553,14 @@ def execute(self, rpc_cmd, **kvargs):
         """
 
         if self.connected is not True:
-            raise EzErrors.ConnectClosedError(self)
+            if self.autoreconnect > 0:
```

Comment:
I don't think we should directly modify the `autoreconnect` value in this block. If `execute()` raises some exception such as a `RpcError` and the caller catches the exception, then we can end up in a situation where the value of `autoreconnect` get decremented, but never incremented.